trino/serial.go

```diff
@@ -35,6 +35,65 @@ func (e UnsupportedArgError) Error() string {
 // If another string format is used it will error to serialise
 type Numeric string
 
+// trinoDate represents a Date type in Trino.
+type trinoDate struct {
+	year  int
+	month time.Month
+	day   int
+}
+
+// Date creates a representation of a Trino Date type.
+func Date(year int, month time.Month, day int) trinoDate {
+	return trinoDate{year, month, day}
+}
+
+// trinoTime represents a Time type in Trino.
+type trinoTime struct {
+	hour       int
+	minute     int
+	second     int
+	nanosecond int
+}
+
+// Time creates a representation of a Trino Time type. To represent time with precision higher than nanoseconds, pass the value as a string and use a cast in the query.
+func Time(hour int,
+	minute int,
+	second int,
+	nanosecond int) trinoTime {
+	return trinoTime{hour, minute, second, nanosecond}
+}
+
+// trinoTimeTz represents a Time(9) With Timezone type in Trino.
+type trinoTimeTz time.Time
+
+// TimeTz creates a representation of a Trino Time(9) With Timezone type.
+func TimeTz(hour int,
+	minute int,
+	second int,
+	nanosecond int,
+	location *time.Location) trinoTimeTz {
+	// When reading a time, a nil location indicates UTC.
+	// However, passing nil to time.Date() panics.
+	if location == nil {
+		location = time.UTC
+	}
+	return trinoTimeTz(time.Date(0, 0, 0, hour, minute, second, nanosecond, location))
+}
+
+// Timestamp indicates we want a TimeStamp type WITHOUT a time zone in Trino from a Golang time.
+type trinoTimestamp time.Time
+
+// Timestamp creates a representation of a Trino Timestamp(9) type.
```

Comment:
those comments are not really useful? @nineinchnick do we have check which enforces those?

Reply:
We don't currently have a linter in the CI that would check for this. The built-in `go vet` doesn't, we'd have to use `staticcheck` (recommended after `go lint` was deprecated) or `golangci-lint` (which is huge but popular).

Having comments for all exported functions is idiomatic in Go and it would show up in online docs like here: https://pkg.go.dev/github.com/trinodb/trino-go-client@v0.308.0/trino

What would it take to make them more useful? Should it explain why `time.Time` is not used instead?

Reply:
> What would it take to make them more useful

Not sure. Presonally I would just drop it. But if we want to follow Go guildelines let's go with it :)